shared/node/node.model.ts

```diff
@@ -0,0 +1,38 @@
+import { Schema, model, Document } from 'mongoose';
+import { Node, NodeType } from './node.interface';
+
+import config from '../../config';
+
+const { collectionNames } = config.mongo;
+
+const nodeSchema = new Schema({
+    id: {
+        type: String,
+        required: true,
+        unique: true,
+        index: true,
+    },
+    name: {
+        type: String,
+        required: true,
+        unique: true,
+        index: true,
+    },
+    type: {
+        type: String,
+        enum: NodeType,
+        default: NodeType.Unknown,
+    },
+    nodeGroupId: {
```

Comment:
```suggestion
    nodesGroupId: {
```

spelling according to the interface